packages/cpt-ui/__tests__/AccessProvider.test.tsx

```diff
@@ -128,4 +148,196 @@ describe("AccessProvider", () => {
+
+      renderWithProvider()
+
+      act(() => {
+        jest.advanceTimersByTime(15001)
+      })
+
+      expect(logger.debug).toHaveBeenCalledWith(
+        "Not checking user info"
+      )
+      expect(mockUpdateTrackerUserInfo).not.toHaveBeenCalled()
+    })
+
+    it("should skip user info check when on allowed no-role paths", async () => {
+      // This test focuses on the isSigningIn logic since we can't easily mock window.location
+      ;(mockUseAuth as jest.Mock).mockReturnValue({
+        isSignedIn: true,
+        isSigningIn: true, // This will trigger the skip logic
+        updateTrackerUserInfo: mockUpdateTrackerUserInfo
+      })
+      ;(useLocation as jest.Mock).mockReturnValue({pathname: FRONTEND_PATHS.LOGIN})
+
+      renderWithProvider()
+
+      act(() => {
+        jest.advanceTimersByTime(15001)
+      })
+
+      expect(logger.debug).toHaveBeenCalledWith(
+        "Not checking user info")
+      expect(mockUpdateTrackerUserInfo).not.toHaveBeenCalled()
+    })
+
+    it("should call updateTrackerUserInfo when user is signed in and not on restricted paths", async () => {
+      mockUpdateTrackerUserInfo.mockResolvedValue({error: null})
+
+      ;(mockUseAuth as jest.Mock).mockReturnValue({
+        isSignedIn: true,
+        isSigningIn: false,
+        updateTrackerUserInfo: mockUpdateTrackerUserInfo
+      })
+      ;(useLocation as jest.Mock).mockReturnValue({pathname: "/search-by-prescription-id"})
+
+      renderWithProvider()
+
+      await act(async () => {
+        jest.advanceTimersByTime(15001)
+      })
+
+      expect(logger.info).toHaveBeenCalledWith("Periodic user info check")
+      expect(logger.info).toHaveBeenCalledWith("Refreshing user info")
+      expect(mockUpdateTrackerUserInfo).toHaveBeenCalled()
+    })
+
+    it("should navigate to session logged out page when updateTrackerUserInfo returns error", async () => {
+      mockUpdateTrackerUserInfo.mockResolvedValue({error: "Session expired"})
+
+      ;(mockUseAuth as jest.Mock).mockReturnValue({
+        isSignedIn: true,
+        isSigningIn: false,
+        updateTrackerUserInfo: mockUpdateTrackerUserInfo
+      })
+      ;(useLocation as jest.Mock).mockReturnValue({pathname: "/search-by-prescription-id"})
+
+      renderWithProvider()
+
+      await act(async () => {
+        jest.advanceTimersByTime(15001)
+      })
+
+      expect(mockUpdateTrackerUserInfo).toHaveBeenCalled()
+      expect(navigate).toHaveBeenCalledWith(FRONTEND_PATHS.SESSION_LOGGED_OUT)
+    })
+
+    it("should not call updateTrackerUserInfo when user is not signed in", async () => {
+      ;(mockUseAuth as jest.Mock).mockReturnValue({
+        isSignedIn: false,
+        isSigningIn: false,
+        updateTrackerUserInfo: mockUpdateTrackerUserInfo
+      })
+      ;(useLocation as jest.Mock).mockReturnValue({pathname: "/cookies"})
+
+      renderWithProvider()
+
+      await act(async () => {
+        jest.advanceTimersByTime(15001)
```

Comment:
These tests will need to be updated to `300001` milliseconds (unless we update the tests to be agnostic about timer duration)